### PR TITLE
Rebase of sd/improvements

### PR DESCRIPTION
**Pull request by sjkelly**
Also try to make CI pass so we can get closer to publishing this.

### CODE REVIEW

**Comment by SimonDanisch** [comment]
I don't really know why appveyor isn't picking up this PR, but it seems to work fine for master...

**Comment by SimonDanisch** [comment]
I'm inclined to merge this and tag FileIO. I'm not sure if I forgot anything, but most of the problems should now be in external packages. [Images](https://github.com/timholy/Images.jl/pull/352) will take some more time but if we tag now we can at least advance with Meshes.

File: src/query.jl [hunk 1]
```diff
@@ -254,7 +258,7 @@ function query(filename::AbstractString)
     if haskey(ext2sym, ext)
         sym = ext2sym[ext]
         len = lenmagic(sym)
-        if length(len) == 1 && (all(x->x==0, len) || !isfile(filename)) # we only found one candidate and there is no magic bytes, or no file, trust the extension
+        if length(len) == 1 && (((all(x->x==0, len) || !isfile(filename))) || !isfile(filename)) # we only found one candidate and there is no magic bytes, or no file, trust the extension
```
Comment (sjkelly) [inline src/query.jl]: Parenthesis are not needed here.

File: src/query.jl [hunk 2]
```diff
@@ -313,8 +318,13 @@ seekable(::Any) = false
 
 function iter_eq(A, B)
     length(A) == length(B) || return false
-    for (a,b) in zip(A,B)
-        a == b || return false
+    i,j = 1,1
+    for _=1:length(A)
```
Comment (sjkelly) [inline src/query.jl]: `eachindex` is a faster alternative on 0.4

Comment (sjkelly) [inline src/query.jl]: Nevermind, this is also a 0.3 package.

**Comment by sjkelly** [comment]
@timholy any objections to merging this? I think we are antsy to get MeshIO published. Semver should allow us to tag and if we need to make any API changes it can bump the minor version. Anything that gets registered can come in as a patch version bump.

File: src/query.jl [hunk 3]
```diff
@@ -254,7 +258,7 @@ function query(filename::AbstractString)
     if haskey(ext2sym, ext)
         sym = ext2sym[ext]
         len = lenmagic(sym)
-        if length(len) == 1 && (all(x->x==0, len) || !isfile(filename)) # we only found one candidate and there is no magic bytes, or no file, trust the extension
+        if length(len) == 1 && ((all(x->x==0, len) || !isfile(filename)) || !isfile(filename)) # we only found one candidate and there is no magic bytes, or no file, trust the extension
```
Comment (timholy) [inline src/query.jl]: I'm confused about this line: aren't you effectively doing `isfile(filename)` twice?

File: src/query.jl (hunk 2 above)
Comment (timholy) [inline src/query.jl]: Though with Compat, `eachindex` is available.

File: src/registry.jl [hunk 4]
```diff
@@ -129,17 +129,59 @@ add_saver(format"WPG", :ImageMagick)
 # Mesh formats
 
 add_format(format"OBJ", (), ".obj")
-add_loader(format"OBJ", :WavefrontObj)
-add_saver(format"OBJ", :WavefrontObj)
+add_loader(format"OBJ", :MeshIO)
```
Comment (timholy) [inline src/registry.jl]: Not that it much matters, but originally I had this file organized into "simple cases" (no function needed) at the top, and "complex cases" at the bottom. They are now intermingled; might be best to delete those comments.

**Comment by timholy** [comment]
No objections here, merge when ready.

File: src/query.jl (hunk 3 above)
Comment (SimonDanisch) [inline src/query.jl]: Uhm, I thought I cleaned that up... Definitely wrong :D

File: src/registry.jl (hunk 4 above)
Comment (SimonDanisch) [inline src/registry.jl]: no, I like it ordered :D The solution here is to make me read more comments...

**Pull request merged**